azure/storageservice/client.go

```diff
@@ -0,0 +1,136 @@
+package storageservice
+
+import (
+	"encoding/base64"
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/MSOpenTech/azure-sdk-for-go/azure"
+)
+
+const (
+	azureStorageServiceListURL = "services/storageservices"
+	azureStorageServiceURL     = "services/storageservices/%s"
+
+	azureXmlns = "http://schemas.microsoft.com/windowsazure"
+
+	errBlobEndpointNotFound = "Blob endpoint was not found in storage serice %s"
+	errParamNotSpecified    = "Parameter %s is not specified."
+)
+
+//NewClient is used to instantiate a new StorageServiceClient from an Azure client
+func NewClient(self *azure.Client) *StorageServiceClient {
+	return &StorageServiceClient{client: self}
+}
+
+func (self *StorageServiceClient) GetStorageServiceList() (*StorageServiceList, error) {
+	storageServiceList := new(StorageServiceList)
+
+	response, err := self.client.SendAzureGetRequest(azureStorageServiceListURL)
+	if err != nil {
+		return nil, err
+	}
+
+	err = xml.Unmarshal(response, storageServiceList)
+	if err != nil {
+		return storageServiceList, err
+	}
+
+	return storageServiceList, nil
+}
+
+func (self *StorageServiceClient) GetStorageServiceByName(serviceName string) (*StorageService, error) {
+	if len(serviceName) == 0 {
+		return nil, fmt.Errorf(errParamNotSpecified, "serviceName")
+	}
+
+	storageService := new(StorageService)
+	requestURL := fmt.Sprintf(azureStorageServiceURL, serviceName)
+	response, err := self.client.SendAzureGetRequest(requestURL)
+	if err != nil {
+		return nil, err
+	}
+
+	err = xml.Unmarshal(response, storageService)
+	if err != nil {
+		return nil, err
+	}
+
+	return storageService, nil
+}
+
+func (self *StorageServiceClient) GetStorageServiceByLocation(location string) (*StorageService, error) {
```

Comment:
more #38, client side logic... I wonder why it exists in the first place. Name is actually misleading. It returns the first storage account found in location. Unless caller knows what order storage accounts are returned (in fact this depends on if the GetStorageServiceList has a special logic or not; otherwise depends on REST API; which probably doesn't define such a contract explicitly) this is dangerous.